includes/class-kadence-blocks-prebuilt-library-rest-api.php

```diff
@@ -588,6 +607,71 @@ public function get_remaining_credits( $request ) {
 			return rest_ensure_response( $response );
 		}
 	}
+	/**
+	 * Handle the event and maybe send to Prophecy WP.
+	 */
+	public function handle_event( $request ) {
+		$event_label = $request->get_param( self::PROP_EVENT_LABEL );
+		$event_data  = $request->get_param( self::PROP_EVENT_DATA );
+		$event_data  = json_decode( $event_data, true );
+
+		$event       = '';
+		$context     = array();
+
+		switch ( $event_label ) {
+			case 'ai_wizard_started':
+				$event = 'AI Wizard Started';
+				break;
+
+			case 'ai_wizard_update':
+				$event = 'AI Wizard Update';
+				$context = [
+					'organization_type' => $event_data['entityType'] ?? '',
+					'location_type'     => $event_data['locationType'] ?? '',
+					'location'          => $event_data['location'] ?? '',
+					'industry'          => $event_data['industry'] ?? '',
+					'mission_statement' => $event_data['missionStatement'] ?? '',
+					'keywords'          => $event_data['keywords'] ?? '',
+					'tone'              => $event_data['tone'] ?? '',
+					'collections'       => $event_data['customCollections'] ?? '',
+				];
+				break;
+			case 'ai_wizard_complete':
+				$event = 'AI Wizard Complete';
+				$context = [
+					'organization_type' => $event_data['entityType'] ?? '',
+					'location_type'     => $event_data['locationType'] ?? '',
+					'location'          => $event_data['location'] ?? '',
+					'industry'          => $event_data['industry'] ?? '',
+					'mission_statement' => $event_data['missionStatement'] ?? '',
+					'keywords'          => $event_data['keywords'] ?? '',
+					'tone'              => $event_data['tone'] ?? '',
+					'collections'       => $event_data['customCollections'] ?? '',
+				];
+				break;
+			case 'pattern_added_to_page':
+				$event = 'Pattern Added to Page';
+				$context = [
+					'pattern_id'         => $event_data['id'] ?? '',
+					'pattern_slug'       => $event_data['slug'] ?? '',
+					'pattern_name'       => $event_data['name'] ?? '',
+					'pattern_style'      => $event_data['style'] ?? '',
+					'pattern_is_ai'      => $event_data['is_ai'] ?? false,
+					'pattern_context'    => $event_data['context'] ?? '',
+					'pattern_categories' => $event_data['categories'] ?? [],
+				];
+				break;
+		}
+
+		if ( strlen($event) !== 0 ) {
+			do_action( 'stellarwp/analytics/event', $event, $context );
+
+			return new WP_REST_Response( [ 'message' => 'Event handled.' ], 200 );
+		}
+
+		return new WP_REST_Response( array( 'message' => 'Event not handled.' ), 200 );
```

Comment:
This should not return a `200` success response, `500` is probably best.

Reply:
Agreed, good idea